thirdparty/faiss/faiss/IVFlib.cpp

```diff
@@ -508,6 +508,7 @@ void ivf_residual_add_from_flat_codes(
                 // ok
                 index->rq.decode(tmp_code.data(), tmp.data(), 1);
                 float norm = fvec_norm_L2sqr(tmp.data(), rq.d);
+                norm = (norm == 0.0 ? 1.0 : norm);
```

Comment:
no, because it uses norm for `distance = x^2 - 2xy + y^2` calculation, not for a cosine-related code